File: p1/app/src/main/java/com/example/logdemoa/MainActivity.java
package com.example.logdemoa;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.util.Log;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        //Log 추가
        Log.e("tag", "error message");
        Log.w("tag", "warning message");
        Log.i("tag", "information message");
        Log.d("tag", "debugging message");
        Log.v("tag", "verbose message");
    }
}

File: ForegroundService/app/src/main/java/com/example/foregroundservice/MainActivity.kt
package com.example.foregroundservice

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        btn_start.setOnClickListener {
            val intent = Intent(this@MainActivity, MusicPlayerService::class.java)
            intent.action = "com.gold24park.musicplayerexample.action.startforeground"
            startService(intent)
        }

        btn_stop.setOnClickListener {
            val intent = Intent(this@MainActivity, MusicPlayerService::class.java)
            intent.action = "com.gold24park.musicplayerexample.action.stopforeground"
            startService(intent)
        }
    }

}
